src/Analysis/Ast/Impl/Analyzer/Symbols/FunctionEvaluator.cs

```diff
@@ -119,10 +225,10 @@ public FunctionEvaluator(ExpressionEval eval, PythonFunctionOverload overload)
             var value = Eval.GetValueFromExpression(node.Expression);
             if (value != null) {
                 // although technically legal, __init__ in a constructor should not have a not-none return value
-                if (FunctionDefinition.Name.EqualsOrdinal("__init__") && _function.DeclaringType.MemberType == PythonMemberType.Class 
-                    && !value.IsOfType(BuiltinTypeId.NoneType)) { 
+                if (FunctionDefinition.Name.EqualsOrdinal("__init__") && _function.DeclaringType.MemberType == PythonMemberType.Class
```

Comment:
See https://github.com/microsoft/python-language-server/issues/1401. You may want to make an extension to check it is `__init__` constructor.

Reply:
Ok